File: RB101_Programming_Foundations/Small Probems/Easy4/multiples_of_3_5.rb
#input: a number
# find multiples of 3 or 5 between 1 and the given number
# compute the sum of those numbers
#output: return the sum


# def multisum(number)
#   sum = 0
#   for i in 1..number
#     if i % 3 == 0 || i % 5 == 0
#       sum += i
#     end
#   end
#   sum
# end

def multisum(number)

  1.upto(number).inject(0) do |acc, num|
    if num % 3 == 0 || num % 5 == 0
      acc += num
    end
    acc
  end
end


puts multisum(3) == 3
puts multisum(5) == 8
puts multisum(10) == 33
puts multisum(1000) == 234168

File: RB101_Programming_Foundations/small_problems/Easy4/Easy4_Python/multiples_of_3_5.py
# Write a method that searches for all multiples of 3 or 5 that lie between 1 and some other number, and then computes the sum of those multiples. For instance, if the supplied number is 20, the result should be 98 (3 + 5 + 6 + 9 + 10 + 12 + 15 + 18 + 20).

# You may assume that the number passed in is an integer greater than 1.

'''
Input : number
output : sum of all numbers between 1 and the number that are multiples of 3 or 5

ALgo :
    start a sum at 0
    iterte from 1 to the number
    check if the current number is divsible by 3 or 5
        if so, add it to the sum
    reutnr sum


'''

def multisum(number):
    sum = 0
    for num in range(1, number + 1):
        if num % 3 == 0 or num % 5 == 0:
            sum += num

    return sum



print(multisum(3) == 3)
print(multisum(5) == 8)
print(multisum(10) == 33)
print(multisum(1000) == 234168)